cpp/arcticdb/column_store/memory_segment.hpp

```diff
@@ -389,10 +389,10 @@ class SegmentInMemory {
         impl_->set_string_pool(string_pool);
     }
 
-    SegmentInMemory filter(util::BitSet&& filter_bitset,
+    SegmentInMemory filter(util::BitSet filter_bitset,
                            bool filter_down_stringpool=false,
                            bool validate=false) const{
-        return SegmentInMemory(impl_->filter(std::move(filter_bitset), filter_down_stringpool, validate));
+        return SegmentInMemory(impl_->filter(filter_bitset, filter_down_stringpool, validate));
```

Comment:
Wouldn't this copy the bit set twice. Once when this function is called and once when impl_->filter is called? I see that impl_->filter modifies the bitset so we can't const ref it, but we could at least std::move to impl_->filter.

Reply:
Yep you're right, only needs copying once in the outermost loop